pkg/devcontainer/runner.go

```diff
@@ -157,3 +135,34 @@ func (r *DockerRunner) executeLifecycleCommandInContainer(ctx context.Context, l
 
 	return nil
 }
+
+func (r *DockerRunner) pullOrBuildImage(ctx context.Context, config Config, projectPath string) (string, error) {
```

Comment:
Hm, we do more that that. Maybe we should just abstract `case config.IsImageDevContainer()`

Reply:
extracted the `IsImageDevContainer` case in a separate method. we can abstract the whole image thing further into something like `ImageProvider` but i think it's an overkill for now.